### PR TITLE
[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29494 [quant][graphmode] Returning axis from calculate_qparams
* #29493 [quant][graphmode] Test for per channel graph mode quantization
* **#29416 [quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`**

Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580906](https://our.internmc.facebook.com/intern/diff/D18580906)

### CODE REVIEW

**Review by ZolotukhinM — APPROVED** [review]
Looks good!

File: test/test_quantization.py [hunk 1]
```diff
@@ -766,30 +768,38 @@ def test_observer_with_ignored_function(self):
...
+            # compare the result of the two quantized models later
+            linear_model.fc1.weight = torch.nn.Parameter(annotated_linear_model.fc1.module.weight.detach())
+            linear_model.fc1.bias = torch.nn.Parameter(annotated_linear_model.fc1.module.bias.detach())
+            model_eager = quantize(annotated_linear_model, test_only_eval_fn,
+                                   self.calib_data)
+
+            qconfig_dict = {
+                '': qconfig
+            }
+            model_script = quantize_script(
```
Comment (raghuramank100) [inline test/test_quantization.py]: If I use prepare_script(), will the output be serializable? I am wondering because part of the observer is in python and part is scripted.

Comment (jerryzh168) [reply test/test_quantization.py]: no, observers with ignored functions are not serializable, do we want to serialize observers?

Comment (jerryzh168) [reply test/test_quantization.py]: oh, looks like they are still serializable..
